src/sentry/api/endpoints/project_group_index.py

```diff
@@ -137,7 +138,10 @@ def _build_query_params_from_request(self, request, project):
 
         query = request.GET.get('query', 'is:unresolved').strip()
         if query:
-            query_kwargs.update(parse_query(project, query, request.user))
+            try:
+                query_kwargs.update(parse_query(project, query, request.user))
+            except InvalidQuery:
+                raise ValidationError('malformed query')
```

Comment:
I would probably raise ValidationError with the message you want to show on the frontend. Or alternatively, on InvalidQuery, return a Response with a custom error code. That avoids us parsing a human readable string to see what's going on in API calls.